pkg/datapath/linux/ipsec.go

```diff
@@ -325,10 +325,6 @@ func (n *linuxNodeHandler) enableIPSecIPv4Do(newNode *nodeTypes.Node, nodeID uin
 		return statesUpdated, errs
 	}
 
-	if !n.nodeConfig.EnableIPSecEncryptedOverlay {
-		return statesUpdated, errs
-	}
```

Comment:
@ldelossa Given you removed this, shouldn't it have impacted the results of the `TestEncryptedOverlayXFRMLeaks` integration test? You disable `EnableIPSecEncryptedOverlay` in that test, before checking that the number of XFRM states and policies are updated.

Reply:
Ah, that test is never called. That's why it never fails when I add support for IPv6 :sweat_smile: Will fix.